### DISCUSSION TITLE
APRS repeater path - setup issue.

### DISCUSSION THREAD

**Discussion by SQ7MHX** [jgromes/RadioLib#934] [Q&A]
Hello, 

I would like to setup APRS path e.g WIDE1-1. I was not able to compile the code successfully.  
I would be grateful for help

Best regards 
Szymon

**Comment by jgromes** [comment]
> I was not able to compile the code successfully.

If that's the case, can you show the code that's failing to compile, including the compilation output?

**Reply by SQ7MHX** [reply]
Hi, 

I tried declaration of object by :
AX25Frame frame; and 
state=frame.setRepeaters('WIDE',1,1);

Compilation errors: 
src\arduino\main.cpp:196:11: error: no matching function for call to 'AX25Frame::AX25Frame()'
  196 | AX25Frame frame;
src\arduino\main.cpp:509:30: error: invalid conversion from 'int' to 'char**' [-fpermissive]
  509 |     state=frame.setRepeaters('WIDE',1,1);
lib\RadioLib\src/protocols/AX25/AX25.h:220:33: note:   initializing argument 1 of 'int16_t AX25Frame::setRepeaters(char**, uint8_t*, uint8_t)'
  220 |     int16_t setRepeaters(char** repeaterCallsigns, uint8_t* repeaterSSIDs, uint8_t numRepeaters);
      |                          ~~~~~~~^~~~~~~~~~~~~~~~~
src\arduino\main.cpp:509:37: error: invalid conversion from 'int' to 'uint8_t*' {aka 'unsigned char*'} [-fpermissive]
  509 |     state=frame.setRepeaters('WIDE',1,1);

using "WIDE" instead of 'WIDE':
src\arduino\main.cpp:509:30: error: cannot convert 'const char [5]' to 'char**'
  509 |     state=frame.setRepeaters("WIDE",1,1);

Best regards, 
Szymon

**Reply by jgromes** [reply]
Well first of all you used single-quotes instead of double quotes. Single quotes are `char`, double quotes are string (aka NULL-terminated character array in C).

Next, you seem to not have read the documentation for `setRepeaters`. It's available on the github pages, here's a link to it: https://jgromes.github.io/RadioLib/class_a_x25_frame.html#a7f2d9f4f1ba29d0fd9f9f3f2cf03f797

As you can see, it expects an array fo repeater callsigns (since there can be up to 8 repeaters in AX.25 frame), not just one callsign. So what you really need is the following:

```c++
AX25Frame frame;
char* repeaters[] = { "WIDE" };
frame.setRepeaters((char**)repeaters, 1, 1);
```

Is one repeater the most common use-case for APRS?

**Reply by SQ7MHX** [reply]
Array usage which you described , solve issues in setRepeaters(); 
I think that WIDE1-1, WIDE2-1 path is most common. 
I reviewed the Wiki it looks like I missed something.  
I used code below and it working fine. 
    AX25Frame frameUI("DESTCALL", D_SSID, "SCALL", S_SSID, RADIOLIB_AX25_CONTROL_U_UNNUMBERED_INFORMATION | RADIOLIB_AX25_CONTROL_POLL_FINAL_DISABLED | RADIOLIB_AX25_CONTROL_UNNUMBERED_FRAME,
                      RADIOLIB_AX25_PID_NO_LAYER_3, "Message" );
    char *repeaters[] = {"WIDE1"};
    uint8_t repssid[] = {1};
    state = frameUI.setRepeaters((char **)repeaters, repssid, 1);

Thank you for support
Best regards,
